### PR TITLE
Fix #1892: Officially deprecate package objects extending js.Any.

### LINKED ISSUE #1892: @js.native cannot be applied to package objects

### PR DESCRIPTION
**Pull request by sjrd**

### CODE REVIEW

**Comment by sjrd** [comment]
Review by @nicolasstucki or @gzm0

File: compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala [hunk 1]
```diff
@@ -458,7 +458,10 @@ abstract class PrepJSInterop extends plugins.PluginComponent
 
       val isJSNative = !sym.hasAnnotation(ScalaJSDefinedAnnotation)
 
-      if (isJSNative && !isJSAnonFun &&
+      if (sym.isPackageObjectClass) {
+        reporter.warning(implDef.pos,
+            "package objects inheriting from js.Any are deprecated")
```
Comment (gzm0) [inline compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala]: - Consider adding: "Use a normal object instead."
- Add a full stop at the and
- Consider starting with upper case (our error messages seem to do so, I don't know about scalac)

**Comment by gzm0** [comment]
That's all.

File: compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala (hunk 1 above)
Comment (sjrd) [inline compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala]: So:

    Package objects inheriting from js.Any are deprecated. Use a normal object instead.

?

Comment (gzm0) [inline compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala]: Yep.

**Comment by sjrd** [comment]
Updated.

**Comment by gzm0** [comment]
LGTM

**Comment by nicolasstucki** [comment]
LGTM

**Pull request merged**